synapse/visibility.py

```diff
@@ -130,9 +142,10 @@ def allowed(event: EventBase) -> Optional[EventBase]:
             sender_erased=erased_senders.get(event.sender, False),
         )
 
```

Comment:
I'm confused why the `state_group_edges` table doesn't have a `room_id` column.

Are the `state_group` ID's globally unique across the instance?

```sh
$ psql synapse
# \d+ state_group_edges
                                 Table "public.state_group_edges"
      Column      |  Type  | Collation | Nullable | Default | Storage | Stats target | Description
------------------+--------+-----------+----------+---------+---------+--------------+-------------
 state_group      | bigint |           | not null |         | plain   |              |
 prev_state_group | bigint |           | not null |         | plain   |              |
```

Reply:
> Are the `state_group` ID's globally unique across the instance?

Seems like they are globally unique given these results:

```
$ psql synapse
# SELECT id, room_id from state_groups;
  id  |                room_id
------+---------------------------------------
    1 | !HBehERstyQBxyJDLfR:my.synapse.server
...
    9 | !HoQvsMxpVGgyZFUPAa:my.synapse.server
...
   39 | !wQEjKbwtnwrPdKAdPe:my.synapse.server
...
   49 | !xoQLgpCPqWzLwUfNBi:my.synapse.server
   50 | !BmcsLnKqJuzDKCJXeW:my.synapse.server
```